doc/snmp/snmp-configdb-migration-hld.md

```diff
@@ -0,0 +1,256 @@
+https://github.com/Azure/SONiC/blob/master/doc/snmp/snmp-schema-addition.md
+
+The new SNMP tables are:
+SNMP
+SNMP_COMMUNITY
+SNMP_USER
+
+
+# SNMP Table
+```
+SNMP|LOCATION
+SNMP|CONTACT
+
+admin@switch1:~$ redis-cli -n 4 hgetall "SNMP|LOCATION"
+1) "LOCATION"
+2) "Emerald City"
+
+admin@switch1:~$ redis-cli -n 4 hgetall "SNMP|CONTACT"
+1) "joe"
+2) "joe@contoso.com"
+```
+
+# SNMP_COMMUNITY Table
+```
+SNMP_COMMUNITY|<community>
+
+admin@switch1:~$ redis-cli -n 4 hgetall "SNMP_COMMUNITY|Jack"
+1) "TYPE"
+2) "RW"
+
+```
+
+# SNMP_USER Table
+```
+SNMP_USER|<user>
+
+admin@switch1:~$ redis-cli -n 4 hgetall "SNMP_USER|Travis"
+ 1) "SNMP_USER_ENCRYPTION_TYPE"
+ 2) "AES"
+ 3) "SNMP_USER_AUTH_TYPE"
+ 4) "SHA"
+ 5) "SNMP_USER_ENCRYPTION_PASSWORD"
+ 6) "TravisEncryptPass"
+ 7) "SNMP_USER_AUTH_PASSWORD"
+ 8) "TravisAuthPass"
+ 9) "SNMP_USER_TYPE"
+10) "Priv"
+11) "SNMP_USER_PERMISSION"
+12) "RO"
+```
+
+
+
+# New SNMP CLI Commands
+# Show commands
+```
+admin@switch1:~$ show run snmp -h
+Usage: show run snmp [OPTIONS] COMMAND [ARGS]...
+
+  Show SNMP running configuration
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community  show running configuration snmp community
+  contact    show running configuration snmp contact
+  location   show running configuration snmp location
+  users      show running configuration snmp users
+admin@switch1:~$
+```
+
+show run snmp community
+```
+admin@switch1:~$ show run snmp community
+Community String    Community Type
+------------------  ----------------
+Qi                  RO
+Travis              RO
+Bill                RO
+Jack                RW
+public              RO
+Joker               RW
+admin@switch1:~$
+```
+
+show run snmp contact
+```
+admin@switch1:~$ show run snmp contact
+Contact    Contact Email
+---------  -----------------
+Joe        joe@microsoft.com
+admin@switch1:~$
+```
+
+show run snmp location
+```
+admin@switch1:~$ show run snmp location
+Location
+----------
+Redmond
+admin@switch1:~$
+```
+
+show run snmp users
+```
+admin@switch1:~$ show run snmp users
+User    Type    Auth Type    Auth Password    Encryption Type    Encryption Password
+------  ------  -----------  ---------------  -----------------  ---------------------
+Travis  Priv    SHA          TravisAuthPass   AES                TravisEncryptPass
+admin@switch1:~$
+```
+
+
+# Config Commands
+```
+admin@switch1:~$ sudo config snmp -h
+Usage: config snmp [OPTIONS] COMMAND [ARGS]...
+
+  SNMP configuration tasks
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  community
+  contact
+  location
+  user
+admin@switch1:~$
+```
+
+sudo config snmp community
+```
+admin@switch1:~$ sudo config snmp community -h
+Usage: config snmp community [OPTIONS] COMMAND [ARGS]...
+
+Options:
+  -?, -h, --help  Show this message and exit.
+
+Commands:
+  add  Add snmp community
+  del  Delete snmp community
```

Comment:
Minor suggestion, might be good to have "Add snmp community string"

Reply:
will do this in the PR for the commands.